neonvm/Dockerfile

```diff
@@ -27,7 +34,7 @@ COPY pkg/util pkg/util
 # was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
 # the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
 # by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
-RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -tags=${BUILDTAGS} -o manager neonvm/main.go
```

Comment:
This change loses GOOS and GOARCH, is that intentional?

Reply:
It's somewhat intentional (all the environment variables, etc. need to match in order for `go-chef --cook` to work) but I'm not actually sure what these should be.

Perhaps @cicdteam has more context on what breaks without these? (and, why it wasn't needed for the binaries outside of neonvm?)

Reply:
These variables are came from kubebuilder templates. It can be used for cross compiling (not a multi-platform docker image, just compiling). I don't think we need them at all.

Reply:
Ah ok! that makes a lot of sense - thanks!